### PR TITLE
#165020082 add social login functionalities

### PR DESCRIPTION
**Pull request by Karlmusingo**
#### What does this PR do?
add social login functionalities
#### Description of Task to be completed?
- Login with twitter
- Login with google 
- Login with facebook
#### How should this be manually tested?
`localhost:3000/login`
`localhost:3000/signup`
#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#165020082](https://www.pivotaltracker.com/story/show/165020082)

#### Screenshots (if appropriate)

N/A

#### Questions:

N/A

### CODE REVIEW

**Review by coolbeatz71 — APPROVED** [review]

File: public/index.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>Author Haven</title>
```
Comment (dusmel) [inline public/index.html:22]: I think it will be better to have a dynamic title

File: src/Components/Common/Button/Button.jsx [hunk 2]
```diff
@@ -87,6 +87,7 @@ Button.defaultProps = {
   icon: null,
   loading: false,
   outline: false,
+  onClick: null,
```
Comment (dusmel) [inline src/Components/Common/Button/Button.jsx:90]: Here I think the click on a button is a must and shouldn't be optional

File: src/Components/SocialLogin.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const { accessToken } = response;
+    onSocialLogin(accessToken, provider);
+  };
+
+  /**
+   * @returns {*} dispatch
+   */
+  render() {
+    return (
+      <>
```
Comment (dusmel) [inline src/Components/SocialLogin.jsx]: I don't get what these tags are doing

File: src/Components/SocialLogin.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        <GoogleLogin
+          id="google"
+          clientId={process.env.REACT_APP_GOOGLE_APP_ID}
+          render={renderProps => (
+            <Button social="google" onClick={renderProps.onClick} disabled={renderProps.disabled} />
```
Comment (dusmel) [inline src/Components/SocialLogin.jsx]: I don't understand the need to disable the social button could you remove that?

**Review by dusmel — CHANGES_REQUESTED** [review]

File: src/Components/Common/Button/Button.jsx (hunk 2 above)
Comment (Karlmusingo) [reply src/Components/Common/Button/Button.jsx:90]: This button is used in the forms without the onClick event on it. That why I didn't put it as required

File: src/Components/SocialLogin.jsx (hunk 3 above)
Comment (Karlmusingo) [reply src/Components/SocialLogin.jsx]: These tags are just serving as wrapper. They replace React.Fragment.

File: public/index.html (hunk 1 above)
Comment (Karlmusingo) [reply public/index.html:22]: This title should be handled by everyone in their containers by using document.title in componentDidMount.
`componentDidMount(){
    document.title = "My Page Title"
  }`

Comment (dusmel) [reply public/index.html:22]: okay we will all add it, can you do the same here?

File: src/Components/SocialLogin.jsx (hunk 3 above)
Comment (dusmel) [reply src/Components/SocialLogin.jsx]: It is a new syntax which is not supported by many tools and react documentation advise to use `React.fragment` instead. For more information, you can check this [link](https://reactjs.org/docs/fragments.html)

Comment (Karlmusingo) [reply src/Components/SocialLogin.jsx]: Cool...

**Review by dusmel — APPROVED** [review]

**Pull request merged**